pkg/sql/exec_log.go

```diff
@@ -347,23 +347,7 @@ func (p *planner) maybeLogStatementInternal(
 		// overhead latency: txn/retry management, error checking, etc
 		execOverheadNanos := svcLatNanos - processingLatNanos
 
-		// If the statement was recorded by the stats collector, we can extract
-		// the statement fingerprint ID. Otherwise, we'll need to compute it from the AST.
-		stmtFingerprintID := statsCollector.StatementFingerprintID()
-		if stmtFingerprintID == 0 {
-			repQuery := p.stmt.StmtNoConstants
-			if repQuery == "" {
-				flags := tree.FmtFlags(queryFormattingForFingerprintsMask.Get(&p.execCfg.Settings.SV))
-				f := tree.NewFmtCtx(flags)
-				f.FormatNode(p.stmt.AST)
-				repQuery = f.CloseAndGetString()
-			}
-			stmtFingerprintID = appstatspb.ConstructStatementFingerprintID(
-				repQuery,
-				implicitTxn,
-				p.CurrentDatabase(),
-			)
-		}
+		stmtFingerprintID := p.instrumentation.fingerprintId
```

Comment:
I know somewhere in SQL Stats we recompute fingerprints because the implicitTxn flag may change. Does this matter here? https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sqlstats/sslocal/sql_stats_ingestor.go#L412

Reply:
@dhartunian From my understanding, an implicit transaction may be upgraded into an explicit one after the execution of a statement. The sql_stats_ingestor will recompute the fingerprint when flushing a recorded transaction stats object, which is after all the statements in a transaction have been executed. The change i made above is happening when the exec_log is logging the statement during its execution, so i dont think its possible for the implicit txn state to have changed here. Lmk if what i said here is wrong